assets/js/Viewer.js

```diff
@@ -1,17 +1,21 @@
 import React, { Component } from 'react';
+import Sign from './Sign';
 
 class Viewer extends Component {
   render() {
     return(
-      <ul>
-        {this.props.messages.map( msg => {
-          return(
-            <li key={msg}>{msg}</li>
-          );
+      <div>
+        {this.sortedKeys(this.props.signs).map( (key, index) => {
+          let lines = this.props.signs[key];
+          return <Sign key={key} signId={key} lineOne={lines[0]} lineTwo={lines[1]}></Sign>
         })}
-      </ul>
+      </div>
     );
   }
+
+  sortedKeys(signs) {
+    return Object.keys(signs).slice(0).sort();
```

Comment:
I don't think you need the `.slice(0)` here

Reply:
Ah, yeah. Good catch.